src/settings.cc

```diff
@@ -244,11 +244,23 @@ Napi::Value getTraceSettings(const Napi::CallbackInfo& info) {
     return o;
   }
 
+  // oboe sets sample_source to -1 when it was a
+  // "continue" decision OR when it is a trigger trace.
+
+  // for a continued trace (that has metadata) oboe will set
+  // out->sample_rate = -1;
+  // out->sample_source = -1;
+  // out->token_bucket_rate = -1;
+  // out->token_bucket_capacity = -1;
+
+  // for a trigger trace (that DOES NOT have metadata) oboe will set
+  // out->sample_source = -1;
+
+  // the conditional trio represents a continued trace which has metadata.
+  have_metadata = out.sample_rate == -1 && out.sample_source == -1 && out.token_bucket_rate == -1 && out.token_bucket_capacity == -1;
+
   // if an x-trace was not used by oboe to make the decision then
-  // create metadata. oboe sets sample_source to -1 when it was a
-  // "continue" decision, i.e., the trace was continued using the
-  // supplied x-trace (no trace decision was made).
-  have_metadata = out.sample_source == OBOE_SAMPLE_RATE_SOURCE_CONTINUED;
+  // there is need to create metadata.
```

Comment:
This is something that's changed between AO and NH that I tried to highlight in NH-12591.  As long as the inbound trace context is valid per W3C `traceparent` format, it _should_ be used as the agent's metadata regardless whether liboboe made a continue decision.  The code here seems to only use the inbound trace context only if liboboe made a continue decision.

Reply:
I don't understand.
All logic regarding trace state is handled by the Agent (see: https://github.com/appoptics/appoptics-apm-node/blob/nh-main/lib/w3c-trace-context.js)

Reply:
I'm thinking about the acceptance criteria cases 2 and 5, for example -- if a valid `traceparent` header is sent to the NH instrumented application, does it continue using that as the metadata?

Reply:
@cheempz if you are saying that this code ignores what is set in have_metadata prior [L137](https://github.com/appoptics/appoptics-bindings-node/blob/6215c9f1fabb5485b15977185b54e91e9de34c06/src/settings.cc#L137) then: a) yes; b) this has been the case before the change in this PR; c) there redundancy here because passing xtrace in [L202](https://github.com/appoptics/appoptics-bindings-node/blob/6215c9f1fabb5485b15977185b54e91e9de34c06/src/settings.cc#L202) will result in `out.sample_source` -1, and hence original metadata respected;

Do we want to start refactoring here??

Reply:
> I'm thinking about the [acceptance criteria](https://swicloud.atlassian.net/wiki/spaces/NIT/pages/2325479753/W3C+Trace+Context#Acceptance-Criteria) cases 2 and 5, for example -- if a valid `traceparent` header is sent to the NH instrumented application, does it continue using that as the metadata?

There are four possible states for an incoming HTTP request. All the logic is handled by the agent. See here: https://github.com/appoptics/appoptics-apm-node/pull/191

Reply:
We should address this (can be in a separate ticket) so that the NH agent uses a valid inbound trace context -- regarding c) it's not always that a valid passed in xtrace will result in `out.sample_source` -1 (aka continue decision).  If the passed in xtrace is valid, but there is no `sw` upstream decision from the tracestate header, liboboe will make a start trace decision where `out.sample_source` will likely be 6 instead.

Reply:
> We should address this (can be in a separate ticket) so that the NH agent uses a valid inbound trace context -- regarding c) it's not always that a valid passed in xtrace will result in `out.sample_source` -1 (aka continue decision). If the passed in xtrace is valid, but there is no `sw` upstream decision from the tracestate header, liboboe will make a start trace decision where `out.sample_source` will likely be 6 instead.

This is beyond my understanding. Remember though that all this complexity can be avoided (can be a separate ticket) by reverting back to oboe 10.2.1 and implementing W3CTrace Context so that it does not break API barriers (as in https://github.com/appoptics/appoptics-apm-node/pull/191)